### PR TITLE
1894 SDK presents the Simplified Login Controller itself

### PR DESCRIPTION
**Pull request by zamorastp**

### CODE REVIEW

File: Sources/AccountSDKIOSWeb/Lib/Storage/Keychain/KeychainSessionStorage.swift [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import Foundation
 
 internal class KeychainSessionStorage: SessionStorage {
+    private(set) var accessGroup: String?
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Lib/Storage/Keychain/KeychainSessionStorage.swift:4]: Didnt we update this to a computed property. In the last pr?

**Review by Daniel-Echegaray — COMMENTED** [review]
Looks really good! I added some remarks that would be good to get fixed.

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginManager.swift [hunk 2]
```diff
@@ -97,7 +88,10 @@ extension SimplifiedLoginManager {
             case .success(let fetchedData):
                 DispatchQueue.main.async {
                     let simplifiedLoginViewController = self.makeViewController(clientName, assertionFetcher: fetcher, simplifiedLoginData: fetchedData)
-                    completion(.success(simplifiedLoginViewController))
+                    if let visibleVC = KeyWindow.get()?.visibleViewController {
```
[the remark this answers is not in the document]
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginManager.swift:91]: Good point!

File: Sources/AccountSDKIOSWeb/Lib/Storage/Keychain/KeychainSessionStorage.swift (hunk 1 above)
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Lib/Storage/Keychain/KeychainSessionStorage.swift]: Rebased :)

File: Sources/AccountSDKIOSWeb/Lib/Extensions/UIWindow+VisibleVC.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        case let tabBarController as UITabBarController:
+            return UIWindow.visibleViewController(from: tabBarController.selectedViewController)
+
+        case let presentingViewController where viewController?.presentedViewController != nil:
+            return UIWindow.visibleViewController(from: presentingViewController?.presentedViewController)
+
+        default:
+            return viewController
+        }
+    }
```
[the remark this answers is not in the document]
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Lib/Extensions/UIWindow+VisibleVC.swift:22]: I added UIWindow as an optional parameter. Please take a look

File: Sources/AccountSDKIOSWeb/Client/Client.swift [hunk 4]
```diff
@@ -230,6 +238,30 @@ public class Client: CustomStringConvertible {
         }
     }
     
+    // Used only for Simplified Login feature with additional parameter window
```
Comment (zamorastp) [inline Sources/AccountSDKIOSWeb/Client/Client.swift:241]: I am not happy about this part of code (duplicated). I needed to pass UiWindow to the Client to have the right window to dismiss the Simplified Login UI... On the other hand, i don't want to make window parameter public in methods getLoginSession

Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Client/Client.swift:241]: Yeaa this was a thinker🤔

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Client/Client.swift:241]: I do not see any other solution for now without unnecessary changes in public api

File: Sources/AccountSDKIOSWeb/Client/Client.swift [hunk 5]
```diff
@@ -150,7 +151,14 @@ public class Client: CustomStringConvertible {
...
             }
 
-            self.handleAuthenticationResponse(url: url, completion: completion)
+            self.handleAuthenticationResponse(url: url) { result in
+                DispatchQueue.main.async {
+                    if let vc = window?.rootViewController {
+                        vc.dismiss(animated: true, completion: nil)
+                    }
+                }
+                completion(result)
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Client/Client.swift:160]: Will this not affect all types of login? Even the ones outside of Simplified Login

**Review by Daniel-Echegaray — APPROVED** [review]

**Pull request merged**